diff --git a/src/Fl_win32.cxx b/src/Fl_win32.cxx
index 3ca0744cc..4daedefbb 100644
--- a/src/Fl_win32.cxx
+++ b/src/Fl_win32.cxx
@@ -603,19 +603,19 @@ class Fl_Win32_At_Exit {
 public:
   Fl_Win32_At_Exit() {}
   ~Fl_Win32_At_Exit() {
-    fl_free_fonts(); // do some Windows cleanup
-    fl_cleanup_pens();
-    OleUninitialize();
-    if (fl_graphics_driver) fl_brush_action(1);
-    fl_cleanup_dc_list();
-    // This is actually too late in the cleanup process to remove the
-    // clipboard notifications, but we have no earlier hook so we try
-    // to work around it anyway.
-    if (clipboard_wnd != NULL)
-      fl_clipboard_notify_untarget(clipboard_wnd);
-#if USE_GDIPLUS
-    Fl_GDIplus_Graphics_Driver::shutdown();
-#endif
+//     fl_free_fonts(); // do some Windows cleanup
+//     fl_cleanup_pens();
+//     OleUninitialize();
+//     if (fl_graphics_driver) fl_brush_action(1);
+//     fl_cleanup_dc_list();
+//     // This is actually too late in the cleanup process to remove the
+//     // clipboard notifications, but we have no earlier hook so we try
+//     // to work around it anyway.
+//     if (clipboard_wnd != NULL)
+//       fl_clipboard_notify_untarget(clipboard_wnd);
+// #if USE_GDIPLUS
+//     Fl_GDIplus_Graphics_Driver::shutdown();
+// #endif
   }
 };
 static Fl_Win32_At_Exit win32_at_exit;
